### PR TITLE
also show if a store path is local or if it needs to be substituted

### PR DESCRIPTION
**Pull request by Mic92**
We deprecate isCached in favor of cacheStatus.

### CODE REVIEW

**Comment by Mic92** [comment]
@mergify queue

**Comment by anmonteiro** [comment]
@Mic92 is it possible something's wrong with this new implementation? I'm getting `cacheStatus: "local"` for every package ([example CI run](https://github.com/nix-ocaml/nix-overlays/actions/runs/9560252119/job/26352262291?pr=1509)) and I'm pretty sure those packages are not cached at all.

**Comment by Mic92** [comment]
Can you reproduce this behavior locally? It works for my uses of nix-eval-jobs i.e. in buildbot-nix and this is more or less just what `nix-build --dry-build` returns. Something that I used also in other tools like: https://github.com/Mic92/nix-build-uncached

**Comment by anmonteiro** [comment]
will try soon -- I tried on aarch64-darwin and got bit by https://github.com/nix-community/nix-eval-jobs/issues/301

**Comment by Mic92** [comment]
Yeah, hopefully I can fix this soon. I got a macbook for testing now and one person that fixed this in Lix was offering me help.

**Comment by anmonteiro** [comment]
Maybe you can help me out, this is the line that seems to cause it:

https://github.com/nix-ocaml/nix-ci-build/blob/9a9478893762b2c3a859a3facad43ea52953f750/flake.nix#L31-L34

If I use system `nix`, I get `isCached: false`. if I use this `$PATH` prefix, I get `isCached: true` for things that are definitely not cached.

**Comment by Mic92** [comment]
Odd that nix-eval-jobs links against nix libraries directly and doesn't use any nix from it's PATH.

**Comment by anmonteiro** [comment]
I just realized that, too. I'm not sure why I added that `$PATH` to `postInstall` since it's assumed `nix` will be in the system. I'll try removing it.

**Comment by anmonteiro** [comment]
I guess this is where I got it from: https://github.com/Mic92/nix-fast-build/blob/cfff239d93716e92f6467f8953d8f8c12da1892a/default.nix#L3

but that doesn't seem to be available anymore

**Comment by anmonteiro** [comment]
honestly this is reproducible for me with:

```nix
$ nix run -L github:nix-community/nix-eval-jobs#nix-eval-jobs --  --gc-roots-dir /tmp/tmppixgt7wv --force-recurse --workers 2 --flake /path/to/my#derivation --check-cache-status
```

File: src/drv.cc [hunk 1]
```diff
@@ -35,7 +36,13 @@ static bool queryIsCached(nix::Store &store,
 
     store.queryMissing(toDerivedPaths(paths), willBuild, willSubstitute,
                        unknown, downloadSize, narSize);
-    return willBuild.empty() && unknown.empty();
+    if (willBuild.empty() && unknown.empty()) {
+        return Drv::CacheStatus::NotBuild;
```
Comment (anmonteiro) [inline src/drv.cc:40]: I think this conditional is just reversed:

before:

`willBuild.empty() && unknown.empty();` -> isCached == true

after:

`willBuild.empty() && unknown.empty();` -> Cache.NotBuild

Comment (Mic92) [reply src/drv.cc:40]: ok. Could you make a pull request?

Comment (anmonteiro) [reply src/drv.cc:40]: working on it

Comment (anmonteiro) [reply src/drv.cc:40]: opened https://github.com/nix-community/nix-eval-jobs/pull/311

**Pull request merged**